plugins/plugin-chart-echarts/src/Tree/transformProps.ts

```diff
@@ -0,0 +1,190 @@
+    const children: TreeSeriesNodeItemOption[][] = [];
+    const indexMap: { [name: string]: number } = {};
+
+    if (!rootNodeName) {
+      return tree;
+    }
+
+    // index indexMap with node ids
+    for (let i = 0; i < data.length; i += 1) {
+      const nodeId = data[i][id] as number;
+      indexMap[nodeId] = i;
+      children[i] = [];
+    }
+
+    // generate tree
+    for (let i = 0; i < data.length; i += 1) {
+      const node = data[i];
+      if (node[parent] === rootNodeId) {
+        tree.children?.push({
+          name: node[nameColumn],
+          children: children[i],
+          value: node[metricLabel],
+        });
+      } else {
+        const parentId = node[parent];
+        if (data[indexMap[parentId]]) {
+          const parentIndex = indexMap[parentId];
+          children[parentIndex].push({
+            name: node[nameColumn],
+            children: children[i],
+            value: node[metricLabel],
+          });
+        }
+      }
+    }
+
+    return tree;
+  }
+
+  let finalTree = {};
+
+  if (rootNodeId) {
+    finalTree = createTree(rootNodeId);
+  } else {
+    /*
+      to select root node,
+      1.find parent nodes with only 1 child.
+      2.build tree for each such child nodes as root
+      3.select tree with most children
+    */
+    // create map of parent:children
+    const parentChildMap: { [name: string]: { [name: string]: any } } = {};
+    data.forEach(node => {
+      const parentId = node[parent] as string;
+      if (parentId in parentChildMap) {
+        parentChildMap[parentId].push({ id: node[id] });
+      } else {
+        parentChildMap[parentId] = [{ id: node[id] }];
+      }
+    });
+
+    // for each parent node which has only 1 child,find tree and select node with max number of children.
+    let maxChildren = 0;
+    Object.keys(parentChildMap).forEach(key => {
+      if (parentChildMap[key].length === 1) {
+        const tree = createTree(parentChildMap[key][0].id);
+        const totalChildren = getTotalChildren(tree);
+        if (totalChildren > maxChildren) {
+          maxChildren = totalChildren;
+          finalTree = tree;
+        }
+      }
+    });
+  }
+
+  const series: TreeSeriesOption[] = [
+    {
+      type: 'tree',
+      data: [finalTree],
+      label: { ...DEFAULT_TREE_SERIES_OPTION.label, position: nodeLabelPosition },
+      emphasis: { focus: emphasis },
+      animation: DEFAULT_TREE_SERIES_OPTION.animation,
+      layout,
+      orient,
+      symbol,
+      roam,
+      symbolSize,
+      lineStyle: DEFAULT_TREE_SERIES_OPTION.lineStyle,
+      select: DEFAULT_TREE_SERIES_OPTION.select,
+      leaves: { label: { position: childLabelPosition } },
+    },
+  ];
+
+  const echartOptions: EChartsOption = {
+    animationDuration: DEFAULT_TREE_SERIES_OPTION.animationDuration,
+    animationEasing: DEFAULT_TREE_SERIES_OPTION.animationEasing,
+    series,
+    tooltip: {
+      trigger: 'item',
+      triggerOn: 'mousemove',
+    },
+  };
```

Comment:
A thought that came up when testing (no need to address this now): The default tooltip formatter appears to join all nodes together:
![image](https://user-images.githubusercontent.com/33317356/117108089-59961700-ad8b-11eb-8b78-d29f6b3f3672.png)
To save space, I think a better layout could be one where a line break is added between node names, not a period. Alternatively, this could be configurable with a control.

Reply:
Will address this seperately.